File: DigitalRSVP.Core/Models/RSVP.cs
namespace DigitalRSVP.Core.Models
{
    public class RSVP
    {
        private Guid _id;
        public Guid Id
        {
            get
            {
                return _id;
            }
            set
            {
                _id = value;
            }
        }

        private Guid _eventId;
        public Guid EventId
        {
            get
            {
                return this._eventId;
            }
            set
            {
                this._eventId = value;
            }
        }

        private Guid _inviteeId;
        public Guid InviteeId
        {
            get
            {
                return _inviteeId;
            }
            set
            {
                _inviteeId = value;
            }
        }

        private DateTime _datetime;
        public DateTime DateTime
        {
            get
            {
                return _datetime;
            }
            set
            {
                if (value > DateTime.Now.AddMinutes(-10))
                {
                    _datetime = value;
                }
            }
        }

        private IEnumerable<Guest>? _guests;
        public IEnumerable<Guest>? Guests
        {
            get
            {
                return this._guests;
            }
            set
            {
                this._guests = value;
            }
        }

        private bool _attendingWedding;
        public bool AttendingWedding
        {
            get
            {
                return this._attendingWedding;
            }
            set
            {
                this._attendingWedding = value;
            }
        }

        private bool _attendingReception;
        public bool AttendingReception
        {
            get
            {
                return this._attendingReception;
            }
            set
            {
                this._attendingReception = value;
            }
        }

        private string? _note;
        public string? Note
        {
            get
            {
                return this._note;
            }
            set
            {
                this._note = value;
            }
        }

        private DateTime _createdDate;
        public DateTime Created_Date
        {
            get
            {
                return this._createdDate;
            }
            set
            {
                if (value > new DateTime(2025, 1, 1))
                {
                    this._createdDate = value;
                }
            }
        }

        private DateTime _updatedDate;
        public DateTime Updated_Date
        {
            get
            {
                return this._updatedDate;
            }
            set
            {
                if (value > new DateTime(2025, 1, 1))
                {
                    this._createdDate = value;
                }
            }
        }

        public RSVP() { }
        public RSVP(Guid id)
        {
            this._id = id;
            this._datetime = DateTime.Now;
        }
    }
}

File: DigitalRSVP.App/src/app/data/rsvp.ts
import { Guest } from "./guest";

export class RSVP {
    private _id: string = '';
    private _eventId: string = '';
    private _inviteeId: string = '';
    private _datetime: Date | null = null;
    private _guests: Array<Guest> | null = null;
    private _note: string = '';
    private _createdDate: Date | null = null;
    private _updatedDate: Date | null = null;

    constructor() {
        this._guests = new Array<Guest>();
    }

    get id(): string {
        return this._id;
    }
    set id(value: string) {
        this._id = value;
    }

    get eventId(): string {
        return this._eventId;
    }
    set eventId(value: string) {
        this._eventId = value;
    }

    get inviteeId(): string {
        return this._inviteeId;
    }
    set inviteeId(value: string) {
        this._inviteeId = value;
    }

    get dateTime(): Date | null {
        return this._datetime;
    }
    set dateTime(value: Date) {
        this._datetime = value;
    }

    get guests(): Array<Guest> | null {
        return this._guests;
    }
    set guests(value: Array<Guest>) {
        this._guests = value;
    }

    get note(): string {
        return this._note;
    }
    set note(value: string) {
        this._note = value;
    }

    get createdDate(): Date | null {
        return this._createdDate;
    }
    set createdDate(value: Date) {
        this._createdDate= value;
    }

    get updatedDate(): Date | null { 
        return this._updatedDate;
    }
    set updatedDate(value: Date) {
        this._updatedDate = value;
    }

    SerializeToJson(): string {
        let json = {
            Id: this._id,
            EventId: this._eventId,
            InviteeId: this._inviteeId,
            DateTime: this._datetime,
            Guests: this._guests,
            Note: this._note
        };
        if (this._guests) {
            this._guests.forEach(guest => {
                let index = json.Guests!.indexOf(guest);
                json.Guests![index] = JSON.parse(guest.SerializeToJson());
            });
        }
        return JSON.stringify(json);
    }
}